doc/xrcvd/transceiver-monitor-hld.md

```diff
@@ -163,16 +334,103 @@ Another entPhySensorTable which is defined in [Entity Sensor MIB(RFC3433)](https
-
-      
+More detailed information about new table and new OIDs are described in [Sensor and Transceiver Info Monitoring Requirement](https://github.com/Azure/SONiC/blob/master/doc/xrcvd/OIDsforSensorandTransciver.MD#transceiver-requirements-entity-mib).
+
+## 3. CLI change ##
+
+### 3.1 Add a command to fetch the error status ###
+
+#### 3.1.1 sfputil extension ####
+
+The utility `sfputil` needs to be updated to extend to support fetching error status of SFP modules.
+The CLI is like this:
+
+    sfputil show error-status --fetch-from-hardware --port <port>
+
+The error status can be fetched for
+
+- A specific SFP module designated by --port argument
+- or all SPF modules if --port isn't provided
+
+By default, it will fetch the error status from `TRANSCEIVER_STATUS` in `STATE_DB`. It also supports fetching error status from low level component directly. In this case, it will call platform API `get_error_description` from the `pmon` docker.
+
+The error status of each SPF module should be:
+
+- `unplugged` if the module isn't plugged-in
+- `OK` if the module is plugged in without any error
+- Errors stored in `STATE_DB` or fetched from platform API.
+
+The output of the command is like this:
+
+    admin@sonic:~# sfputil show error-status --port Ethernet8
+    Port       Error Status
+    ---------  ------------------------------------
+    Ethernet8  OK
+
+    admin@sonic:~# sfputil show error-status
+    Port         Error Status
+    -----------  ----------------------------------------------
+    Ethernet0    OK
+    Ethernet4    OK
+    Ethernet8    OK
+    Ethernet12   OK
+    Ethernet16   OK
+    Ethernet20   OK
+    Ethernet24   OK
+    Ethernet28   OK
+    Ethernet32   OK
+    Ethernet36   OK
+    Ethernet40   OK
+    Ethernet44   Power budget exceeded
+    Ethernet48   OK
+    Ethernet52   OK
+    Ethernet56   OK
+    Ethernet60   OK
+    Ethernet64   OK
+    Ethernet68   OK
+    Ethernet72   OK
+    Ethernet76   OK
+    Ethernet80   OK
+    Ethernet84   OK
+    Ethernet88   OK
+    Ethernet92   OK
+    Ethernet96   OK
+    Ethernet100  OK
+    Ethernet104  OK
+    Ethernet108  OK
+    Ethernet112  OK
+    Ethernet116  OK
+    Ethernet120  OK
+    Ethernet124  OK
+    Ethernet128  OK
+    Ethernet132  OK
+    Ethernet136  OK
+    Ethernet140  OK
+    Ethernet144  OK
+    Ethernet148  OK
+    Ethernet152  OK
+    Ethernet156  OK
+    Ethernet160  OK
+    Ethernet164  OK
+    Ethernet168  Unplugged
+    Ethernet172  OK
+    Ethernet176  OK
+    Ethernet180  OK
+    Ethernet184  OK
+    Ethernet188  OK
+    Ethernet192  OK
+    Ethernet196  OK
+    Ethernet200  OK
+    Ethernet204  OK
+    Ethernet208  OK
+    Ethernet212  OK
+    Ethernet216  OK
+    Ethernet220  OK
+
+#### 3.1.2 show interface transceiver wrapper ####
+
+A wrapper command `show interface transceiver error-status` is also provided.
+
+    show interface transceiver error-status --fetch-from-hardware port

```

Comment:
>port [](http://example.com/codeflow?start=66&length=4)

What is `port`? do you mean `--port <port>` ?

Reply:
Hi @qiluo-msft
For `show interface transceiver` command, we don't have `--port` prefix. It is just followed by `port`.
So I don't tend to change this. How do yuo think?
FYI, current subcommands
```

admin@arc-mtbc-1001:~$ show interfaces transceiver eeprom Ethernet0
Ethernet0: SFP EEPROM detected
        Application Advertisement: N/A
        Connector: Unknown
        Encoding: 64B/66B
        Extended Identifier: GBIC/SFP defined by twowire interface ID
        Extended RateSelect Compliance: Unspecified
        Identifier: SFP/SFP+/SFP28
        LengthOM3(UnitsOf10m): 0
        Nominal Bit Rate(100Mbs): 255
        Specification compliance:
                EthernetComplianceCodes: 1000BASE-CX
                InfinibandComplianceCode: 1X Copper Passive
                SFP+CableTechnology: Passive Cable
        Vendor Date Code(YYYY-MM-DD Lot): 2017-01-09 
        Vendor Name: Mellanox
        Vendor OUI: 00-02-c9
        Vendor PN: MCP7F00-A002R
        Vendor Rev: A1
        Vendor SN: MT1702VS15044
admin@arc-mtbc-1001:~$ show interfaces transceiver lpmode Ethernet0
Port       Low-power Mode
---------  ----------------
Ethernet0  On
admin@arc-mtbc-1001:~$ show interfaces transceiver presence Ethernet0
Port       Presence
---------  ----------
Ethernet0  Present
admin@arc-mtbc-1001:~$ 
```

Reply:
OK, to make your syntax consistent, try
```suggestion
    show interface transceiver error-status --fetch-from-hardware <port>
```